src/memtable/mod.rs

```diff
@@ -131,7 +134,11 @@ impl Memtable {
             .fetch_add(item_size, std::sync::atomic::Ordering::AcqRel);
 
         let key = InternalKey::new(item.key.user_key, item.key.seqno, item.key.value_type);
-        self.items.insert(key, item.value);
+        // TODO(ajwerner): Decide what we want to do here. The panic is sort of
+        // extreme, but also seems right given the invariants.
```

Comment:
Write transactions write with a fixed sequence number initially, so we actually need to overwrite values to keep the same behaviour as crossbeam's skiplist

https://github.com/fjall-rs/fjall/blob/5001d4db6430808df4c8ba6db12c2dbaaf7a91ec/src/tx/write/mod.rs#L379-L381

Reply:
FWIW the implementation of the write sets in the transactions *could* be separate from the memtables. That would allow something the like STL B-tree to be used (because transactions are not concurrent anyway), maybe

Then the whole free list stuff would not be necessary, but it would require some code changes somewhere else in lsm-tree

Reply:
This can probably be solved by reserving the 64th (highest) bit in the sequence number for transactions.
Then any write in the transaction starts at 0x80...0. That way the transaction sequence numbers are always higher than the database's.
RocksDB actually one uses 56-bit sequence numbers.